sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileClient.java

```diff
@@ -1135,8 +1139,11 @@ public DataLakeFileOpenInputStreamResult openInputStream(DataLakeFileInputStream
      * @throws DataLakeStorageException If a storage service error occurred.
      */
     public DataLakeFileOpenInputStreamResult openInputStream(DataLakeFileInputStreamOptions options, Context context) {
+        context = BuilderHelper.addUpnHeader(() -> (options == null) ? null : options.isUpn(), context);
+        Context finalContext = context;
```

Comment:
Just a nit, this doesn't do anything useful. Feel free to pass `context` on to the `openInputStream` method. The only time you'd need to do this is if the `context` was used in a lambda as variables used from outside the scope of the lambda have to be either final or effectively final.